src/briefcase/integrations/android_sdk.py

```diff
@@ -126,6 +126,64 @@ def verify_android_sdk(command):
     return sdk
 
 
+def _get_choices_and_names(running_devices, all_emulators, avd_name_by_device_id):
+    '''Input:
+
+    - `running_devices`: Dict from ADB device ID to dict with 'model' property.
+    - `all_emulators`: List of emulator device names.
+    - `avd_name_by_device_id`: A dict from ADB device ID to AVD name.
+
+    Output:
+
+    Tuple of `choice_ids`, `full_names`.
+
+    `choice_ids` is a list of all running devices, followed by all non-running
+    AVD names prefixed with `@`.
+
+    `full_names` is dict from string (device ID *or* AVD) to a formatted description
+    of the device.
+    '''
+    choice_ids = []
+    full_names = {}
+
+    running_avds = set()  # Used to avoid adding an AVD twice to the list of choices.
+
+    # Handle devices that are running.
+    for device_id, device_info in sorted(running_devices.items(), key=lambda d: d[1]["name"]):
+        model = device_info["name"]
+        if avd_name_by_device_id.get(device_id):
+            avd = avd_name_by_device_id[device_id]
+            running_avds.add(avd)
+            # For emulators, the full_name shows the @{avd} name. This is because the
+            # ADB device ID is not stable over reboots for Android virtual devices.
+            printable_device_id = "@{avd}".format(avd=avd)
+            device_type = "emulator"
+        else:
+            printable_device_id = device_id
+            device_type = "device"
+
+        full_name = "{printable_device_id} ({model} {device_type})".format(
+            device_type=device_type, model=model, printable_device_id=printable_device_id,
```

Comment:
The "name" is used in the menu, but it's also used in the context of saying "starting device on {name}"; to that end, "Kogan Agora 9 device" is a more useful "name", with the device ID providing important context. 

To support that, I'd be inclined to list then name the other way around for physical devices - "Kogan_Agora_9 device (KABCDABCDA1513)" for a physical device (i.e., what it is to a human, with the ID only there for disambiguation).

The text for emulators can be simplified, too; "Android_SDK_build_for_x86" doesn't convey anything that "emulator" doesn't already convey. Not sure if I prefer "@beePhone (emulator)" or "Emulator (@beePhone)", though. Thoughts?